cst438_gradebook/src/test/java/com/cst438/JunitTestGradebook.java

```diff
@@ -240,7 +240,173 @@ public void updateAssignmentGrade() throws Exception {
+
+			AssignmentGrade ag = new AssignmentGrade();
+			ag.setAssignment(assignment);
+			ag.setId(1);
+			ag.setScore("");
+			ag.setStudentEnrollment(enrollment);
+
+			// given -- stubs for database repositories that return test data
+			given(assignmentRepository.findById(1)).willReturn(assignment);
+			given(assignmentGradeRepository.findByAssignmentIdAndStudentEmail(1, TEST_STUDENT_EMAIL)).willReturn(ag);
+			given(assignmentGradeRepository.save(any())).willReturn(ag);
+			given(courseRepository.findByCourse_id(40442)).willReturn(course);
+
+			// end of mock data
+			
+			//delete the assignment
+			response = mvc.perform(MockMvcRequestBuilders.delete("/gradebook/1").accept(MediaType.APPLICATION_JSON)).andReturn().getResponse();
+
+			// verify that delete was called for both the assignment and the one empty score associated with it
+			
+			verify(assignmentRepository, times(1)).delete(any());
+			verify(assignmentGradeRepository, times(1)).delete(any());
+			
+		}
+	
+		//test renaming an assignment
+		@Test
+		public void renameAssignment() throws Exception {
+
+			MockHttpServletResponse response;
+
+			// mock database data
+
+			Course course = new Course();
+			course.setCourse_id(TEST_COURSE_ID);
+			course.setSemester(TEST_SEMESTER);
+			course.setYear(TEST_YEAR);
+			course.setInstructor(TEST_INSTRUCTOR_EMAIL);
+			course.setEnrollments(new java.util.ArrayList<Enrollment>());
+			course.setAssignments(new java.util.ArrayList<Assignment>());
+
+			Enrollment enrollment = new Enrollment();
+			enrollment.setCourse(course);
+			course.getEnrollments().add(enrollment);
+			enrollment.setId(TEST_COURSE_ID);
+			enrollment.setStudentEmail(TEST_STUDENT_EMAIL);
+			enrollment.setStudentName(TEST_STUDENT_NAME);
+
+			Assignment assignment = new Assignment();
+			assignment.setCourse(course);
+			course.getAssignments().add(assignment);
+			// set dueDate to 1 week before now.
+			assignment.setDueDate(new java.sql.Date(System.currentTimeMillis() - 7 * 24 * 60 * 60 * 1000));
+			assignment.setId(1);
+			assignment.setName("Assignment 1");
+			assignment.setNeedsGrading(1);
+
+			AssignmentGrade ag = new AssignmentGrade();
+			ag.setAssignment(assignment);
+			ag.setId(1);
+			ag.setScore("");
+			ag.setStudentEnrollment(enrollment);
+
+			// given -- stubs for database repositories that return test data
+			given(assignmentRepository.findById(1)).willReturn(assignment);
+			given(assignmentGradeRepository.findByAssignmentIdAndStudentEmail(1, TEST_STUDENT_EMAIL)).willReturn(null);
+			given(assignmentGradeRepository.save(any())).willReturn(ag);
+			given(courseRepository.findByCourse_id(40442)).willReturn(course);
+
+			// end of mock data
+			
+			//create a new assignment
+			response = mvc.perform(MockMvcRequestBuilders.put("/gradebook/1/Test_Rename").accept(MediaType.APPLICATION_JSON)).andReturn().getResponse();
+
+			// verify that assignment was modified
+			
+			verify(assignmentRepository, times(1)).save(any()); // ???
+
+		}
+	
```

Comment:
Overall the junit tests look good! I don't know if it might be relevant to test the data being returned, evaluating with an if statement that modifies a Boolean and then a statement like

		assertTrue("Assignment was added.", found);

or something like that, but that's what I have going in my junit tests.